File: backend-TFG/src/main/java/es/udc/fic/tfg/backendtfg/recipes/infrastructure/dtos/CreateRecipeParamsDTO.java
package es.udc.fic.tfg.backendtfg.recipes.infrastructure.dtos;

import lombok.*;

import javax.validation.constraints.*;
import java.util.List;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
public class CreateRecipeParamsDTO {
    @NotBlank
    @Size(min = 1, max = 100)
    private String name;
    
    private String description;
    
    @NotNull
    @Positive
    private long duration;
    
    @NotNull
    @Positive
    private int diners;
    
    @NotNull
    private UUID authorID;
    
    @NotNull
    private UUID categoryID;
    
    @NotNull
    private List<CreateRecipeStepParamsDTO> steps;
    
    private List<CreateRecipePictureParamsDTO> pictures;
    
    @NotNull
    private List<CreateRecipeIngredientParamsDTO> ingredients;
}

File: frontend-tfg/src/Recipes/Infrastructure/CreateRecipeParamsDTO.ts
import CreateRecipeStepParamsDTO from "./CreateRecipeStepParamsDTO";
import CreateRecipePictureParamsDTO from "./CreateRecipePictureParamsDTO";
import CreateRecipeIngredientParamsDTO from "./CreateRecipeIngredientParamsDTO";

interface CreateRecipeParamsDTO {
    name: string,
    description: Nullable<string>,
    duration: number,
    diners: number,
    authorID: string,
    categoryID: string,
    steps: Array<CreateRecipeStepParamsDTO>,
    pictures: Nullable<Array<CreateRecipePictureParamsDTO>>,
    ingredients: Array<CreateRecipeIngredientParamsDTO>
}

export default CreateRecipeParamsDTO;
